### ISSUE TITLE
cmake install target does not install bundled libs

### ISSUE THREAD

**Issue by kcris** [dealii/dealii#17767]
the installation step for deal-ii is incomplete. The cmake script only copies 
* deal-ii headers 
* deal-ii lib 
when using `cmake --install`.

The install target in the cmake script should also install/copy the bundled deps as well, not just the deal-ii files.

Deal-II's `install` target should call the install target of all the bundled deps.

Only this way you make sure that a client project can find and link against deal-ii.

Note: tested on windows, vs2022, plain cmake configuration (no options) 
but I do not think this is Windows specific.

**Comment by marcfehling** [comment]
I checked our CI results. To me it looks like they are installed in `<INSTALLATION_FOLDER>/include/deal.II/bundled/`. Can you confirm that they are not there?

**Comment by tamiko** [comment]
@kcris Bundled dependencies are linked into the deal.II library object directly and necessary header files are installed under `/include/deal.II/bundled/`. Would you mind to check that this is the case for you?

**Comment by kcris** [comment]
Sorry, I by deps I meant the `.lib` files files really.

* yes, headers for bundled libraries are! installed. 
* but `.lib`  filesfor bundled libraries are not. 

Next to deal_II.lib (inside the install/lib folder) there should also be `boost_iostreams.lib`, `boost_serialization.lib`, `kokkos.lib`, `muparsers.lib`

**Comment by tamiko** [comment]
@kcris 
> The deal-ii.lib does not include the deps of deal-ii, just deal-ii object files.

No, it does.

We do not compile our bundled dependencies into separate libraries but incorporate (portions of) these dependencies in the deal.II library object itself. We have chosen this path to minimize the impact of bundled dependencies (in particular when accidentally mixed with the same external dependencies).

**Comment by kcris** [comment]
yes, I do observe a linkage error. Without linking expliclitly to boost_serialization.lib I get errors. When I link against it all is good.

This happens when building the static library (.lib) on windows.

**Comment by tamiko** [comment]
@kcris I will need to reproduce this on Windows to understand what's going on. Please attach the full contents of the file `detailed.log` that you can find in the build directory of deal.II.
